packages/invoices/src/OutboundShipment/DetailView/reducer.ts

```diff
@@ -191,14 +189,30 @@ export const reducer = (
             lines[existingLineIdx] = {
               ...lines[existingLineIdx],
               ...line,
+              id: lines[existingLineIdx]?.id || line.id,
               isUpdated: true,
               isDeleted: false,
             };
           } else {
-            line.isCreated = true;
-            line.isUpdated = true;
-            line.isDeleted = false;
-            draft.lines.push(createLine(line, draft, dispatch));
+            const existingDeletedLineIdx = lines.findIndex(
+              ({ isDeleted, itemId }) => itemId === line.itemId && isDeleted
+            );
+
+            if (existingDeletedLineIdx >= 0) {
+              lines[existingDeletedLineIdx] = {
+                ...lines[existingDeletedLineIdx],
+                ...line,
+                id: lines[existingDeletedLineIdx]?.id || line.id,
+                isCreated: true,
+                isDeleted: false,
+                isUpdated: true,
+              };
+            } else {
+              line.isCreated = true;
+              line.isUpdated = true;
+              line.isDeleted = false;
+              draft.lines.push(createLine(line, draft, dispatch));
+            }
```

Comment:
this is pretty verbose - could maybe do something about it but it seems easier to read to just have the verbose if blocks